entity2/entitykeyvalues.cpp

```diff
@@ -0,0 +1,340 @@
+#include "entity2/entitysystem.h"
+#include "entity2/entitykeyvalues.h"
+#include "tier0/logging.h"
+
+// memdbgon must be the last include file in a .cpp file!!!
+#include "tier0/memdbgon.h"
+
+extern CEntitySystem* EntitySystem();
```

Comment:
Wouldn't it be better to define it in entitysystem.h? so you don't end up redefining it everywhere where needed.